email.ts

```diff
@@ -1,8 +1,8 @@
-export * from './smtp/address';
-export * from './smtp/client';
-export * from './smtp/connection';
-export * from './smtp/date';
-export * from './smtp/error';
-export * from './smtp/message';
-export * from './smtp/mime';
-export * from './smtp/response';
+export * from './smtp/address.js';
```

Comment:
can't this be avoided, and files are referred to without the extension? i find it confusing to refer to a file that doesn't exist (yet)

Reply:
yes, if we specify the `--experimental-module-resolution=node` flag, which iirc is going to be removed when work on loaders is complete (which looks to be soon-ish; [loader chaining is in the PR stage](https://github.com/nodejs/node/pull/42623), which was The Big One as far as features went). 

at any rate, the standard is to specify the extension & typescript's official recommendation (for years) has been ["write the module name as you expect it to work at runtime"](https://github.com/Microsoft/TypeScript/issues/18971#issuecomment-334576547) and just trust the compiler to sort it out.

Reply:
on node `v16.4.2` i'm able to run `tsc` and `rollup` without the extensions.

what breaks when the extension is removed?

Reply:
running the code itself (e.g. testing through the source files & not the bundle)